Added initial changes to dropdown

JurjenBiewenga opened the pull request:
Added initial changes to allow the user to change the selected item while the dropdown is closed but selected and allows the user to open the dropdown by pressing down specified in #71.

JurjenBiewenga committed:
Added initial changes to allow the user to change the selected item while the dropdown is closed but selected and allows the user to open the dropdown by pressing down

grokys committed:
Merge branch 'master' into DropdownKeyboardTweaks

grokys commented on src/Avalonia.Controls/DropDown.cs:
```diff
@@ -114,6 +114,30 @@ protected override void OnKeyDown(KeyEventArgs e)
+                            SelectedIndex = 0;
+                        e.Handled = true;
+                    }
+                }
```
This doesn't seem to be the correct behavior (on windows at least) - on windows the up/down arrows, when the dropdown isn't open, cycle through the available items. It's Alt+Down that opens the dropdown.

JurjenBiewenga replied:
I've made the changes however the keydown event for up/down don't appear to propagate to the dropdown, I'm not sure why that happens. As result the changes do not currently work.

JurjenBiewenga committed:
Made the requested changes

grokys committed:
Merge branch 'master' into pr/1166-added-initial-changes-to-dropdown

grokys committed:
Select dropdown item on focus.

Arrow keys within the dropdown are handled by the keyboard navigation handler, so detect when this focuses a new item and select it.

grokys commented:
@JurjenBiewenga I've pushed a commit which fixes selection with arrow keys in the dropdown. Last issue I can see is that on Windows at least, the selection with arrow keys when the dropdown is closed shouldn't wrap.

grokys committed:
Merge branch 'master' into DropdownKeyboardTweaks

grokys committed:
Merge branch 'master' into DropdownKeyboardTweaks
